src/Mod/Draft/draftguitools/gui_snapper.py

```diff
@@ -1330,123 +1410,98 @@ def cancel():
+        snap_modes = param.GetString("snapModes")
+
+        for b in toolbar.actions():
             if len(b.statusTip()) == 0:
                 b.setStatusTip(b.toolTip())
 
-        # restoring states
-        t = Draft.getParam("snapModes", "111111111101111")
-        if t:
-            c = 0
-            for b in [self.masterbutton] + self.toolbarButtons:
-                if len(t) > c:
-                    state = bool(int(t[c]))
-                    b.setChecked(state)
+        # restore toolbar buttons state
+        if snap_modes:
+            for a in toolbar.findChildren(QtGui.QAction):
+                snap = a.objectName()[11:].replace(button_suffix,"")
+                if snap in Gui.Snapper.snaps:
+                    i = Gui.Snapper.snaps.index(snap)
+                    state = bool(int(snap_modes[i]))
+                    a.setChecked(state)
                     if state:
-                        b.setToolTip(b.toolTip() + " (ON)")
+                        a.setToolTip(a.toolTip()+" (ON)")
                     else:
-                        b.setToolTip(b.toolTip() + " (OFF)")
-                    c += 1
-        if not Draft.getParam("showSnapBar", True):
-            self.toolbar.hide()
+                        a.setToolTip(a.toolTip()+" (OFF)")
+
+
+    def get_snap_toolbar(self):
+        """retuns snap toolbar object"""
+        mw = Gui.getMainWindow()
+        if mw:
+            toolbar = mw.findChild(QtGui.QToolBar,"Draft Snap")
+            if toolbar:
+                return toolbar
+        return None
+
 
     def toggleGrid(self):
-        """Run Draft_ToggleGrid."""
-        FreeCADGui.runCommand("Draft_ToggleGrid")
-
-    def saveSnapModes(self):
-        """Save the snap modes for next sessions."""
-        t = ''
-        for b in [self.masterbutton] + self.toolbarButtons:
-            t += str(int(b.isChecked()))
-            if b.isChecked():
-                b.setToolTip(b.toolTip().replace("OFF", "ON"))
-            else:
-                b.setToolTip(b.toolTip().replace("ON", "OFF"))
-        Draft.setParam("snapModes", t)
-
-    def toggle(self, checked=None):
-        """Toggle the snap mode."""
-        if hasattr(self, "toolbarButtons"):
-            if checked is None:
-                self.masterbutton.toggle()
-            elif checked:
-                if hasattr(self, "savedButtonStates"):
-                    for i in range(len(self.toolbarButtons)):
-                        self.toolbarButtons[i].setEnabled(True)
-                        self.toolbarButtons[i].setChecked(self.savedButtonStates[i])
-            else:
-                self.savedButtonStates = []
-                for i in range(len(self.toolbarButtons)):
-                    self.savedButtonStates.append(self.toolbarButtons[i].isChecked())
-                    self.toolbarButtons[i].setEnabled(False)
-        self.saveSnapModes()
+        "toggle FreeCAD Draft Grid"
```

Comment:
Triple quotes and capital letter for docstring. It should end in a dot.
`"""Toggle Draft grid."""`